src/Functions/snowflake.cpp

```diff
@@ -49,24 +44,34 @@ class FunctionDateTimeToSnowflake : public IFunction
         FunctionArgumentDescriptors args{
             {"value", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isDateTime), nullptr, "DateTime"}
         };
-        validateFunctionArgumentTypes(*this, arguments, args);
+        FunctionArgumentDescriptors optional_args{
+            {"epoch", static_cast<FunctionArgumentDescriptor::TypeValidator>(&isNativeUInt), isColumnConst, "UInt*"}
+        };
+        validateFunctionArgumentTypes(*this, arguments, args, optional_args);
 
         return std::make_shared<DataTypeInt64>();
     }
 
     ColumnPtr executeImpl(const ColumnsWithTypeAndName & arguments, const DataTypePtr &, size_t input_rows_count) const override
     {
         const auto & src = arguments[0];
-        const auto & src_column = *src.column;
+        const auto & col_src = *src.column;
+
+        size_t epoch = 0;
+        if (arguments.size() >= 2 && input_rows_count != 0)
+        {
+            const auto & col_epoch = *arguments[1].column;
+            epoch = col_epoch.getUInt(0);
+        }
 
-        auto res_column = ColumnInt64::create(input_rows_count);
-        auto & res_data = res_column->getData();
+        auto col_res = ColumnInt64::create(input_rows_count);
+        auto & res_data = col_res->getData();
 
-        const auto & src_data = typeid_cast<const ColumnUInt32 &>(src_column).getData();
+        const auto & src_data = typeid_cast<const ColumnUInt32 &>(col_src).getData();
         for (size_t i = 0; i < input_rows_count; ++i)
-            res_data[i] = (Int64(src_data[i]) * 1000 - snowflake_epoch) << time_shift;
+            res_data[i] = (Int64(src_data[i]) * 1000 - epoch) << time_shift;
```

Comment:
I see that this is a breaking change, however we have to break either the generator function or the converter functions. So I would say it is okay if we list this as a breaking change.